src/runtime/sys_windows_arm64.s

```diff
@@ -0,0 +1,501 @@
+// Copyright 2018 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+#include "go_asm.h"
+#include "go_tls.h"
+#include "tls_arm64.h"
+#include "textflag.h"
+
+// void runtime·asmstdcall(void *c);
+TEXT runtime·asmstdcall(SB),NOSPLIT|NOFRAME,$0
+	// Save non-volatile registers
+	SUB 	$32, RSP		// SP = SP - 32
+	MOVD	R19, 0(RSP)
+	MOVD	R20, 8(RSP)
+	MOVD	R21, 16(RSP)
+
+	MOVD	R0, R19			// R19 = libcall* (non-volatile)
+	MOVD	RSP, R20		// R20 = SP (non-volatile)
+	MOVD	LR, R21			// R21 = link register
+	
+	// SetLastError(0)
+	WORD	$0xaa1203e1		// MOVD	R18, R1

```

Comment:
Why can't this be expressed in assembly?

Reply:
R18 is a reserved register in arm64 and Go compiler doesn't let you use it. But we need it as the base address of TEB structure is stored here (as per [windows arm64 ABI](https://docs.microsoft.com/en-us/cpp/build/arm64-windows-abi-conventions?view=vs-2019)).

Reply:
Regardless of correctness, it looks very out of place. A comment explaining why this was needed would help future devs.